sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3398,6 +3399,8 @@ class Analyzer(override val catalogManager: CatalogManager)
       SchemaUtils.checkColumnNameDuplication(
         i.userSpecifiedCols, "in the column list", resolver)
 
+      i.setTagValue(USER_SPECIFIED_COLUMNS_RESOLVED, true)
```

Comment:
> Note, this ResolveDefaultColumns rule currently depends on the whole InsertIntoStatement not being analyzed yet. So now it checks that the new TreeNodeTag is unset.

@dtenedor Oh I didn't realize this. Let's skip setting/reading the tag to make it simple.

Reply:
Sure, NP. Removed the tree tag.